src/scripts/DataLayerItem.js

```diff
@@ -96,6 +183,132 @@ class Item {
+  };
+
+  /**
+   * Indicates whether the item is a listener on or off item.
+   *
+   * @returns {Boolean} true if the item is a listener on or off item, false otherwise.
+   */
+  get isListener() {
+    return (this._type === itemType.LISTENER_ON || this._type === itemType.LISTENER_OFF);
+  };
+
 }
 
+const utils = {
+  /**
+   * Determines whether the item configuration matches the constraints.
+   *
+   * @param {ItemConfig} itemConfig The item configuration.
+   * @param {Object} constraints The constraints on the item configuration.
+   * @returns {Boolean} true if the item configuration matches the constraints, false otherwise.
+   * @static
+   */
+  itemConfigMatchesConstraints: function(itemConfig, constraints) {
+    for (let i = 0; i < constraints.properties.length; i++) {
+      const key = constraints.properties[i].key;
+      const type = constraints.properties[i].type;
+      const values = constraints.properties[i].values;
+      const mandatory = constraints.properties[i].mandatory;
+      const configValue = itemConfig[key];
+      const configValueType = typeof configValue;
+      if (mandatory) {
+        if (!configValue || (configValueType !== type) || !utils.valueIsAllowed(configValue, values)) {
+          return false;
+        }
+      } else {
+        if (configValue && ((configValueType !== type) || !utils.valueIsAllowed(configValue, values))) {
+          return false;
+        }
+      }
+    }
+    return constraints.customPropertiesAllowed || !utils.itemConfigHasCustomProperties(itemConfig, constraints);
+  },
+  /**
+   * Determines whether the item configuration has custom properties.
+   *
+   * @param {ItemConfig} itemConfig The item configuration.
+   * @param {Object} constraints The constraints on the item configuration.
+   * @returns {Boolean} true if the item configuration has custom properties, false otherwise.
+   * @static
+   */
+  itemConfigHasCustomProperties: function(itemConfig, constraints) {
+    const itemConfigKeys = Object.keys(itemConfig);
+    if (itemConfigKeys.length > constraints.properties.length) {
+      return true;
+    }
+    for (let j = 0; j < itemConfigKeys.length; j++) {
+      const itemConfigKey = itemConfigKeys[j];
+      let itemConfigKeyMatchesConstraintKey = false;
+      for (let k = 0; k < constraints.properties.length; k++) {
+        const key = constraints.properties[k].key;
+        if (itemConfigKey === key) {
+          itemConfigKeyMatchesConstraintKey = true;
+          break;
+        }
+      }
+      if (!itemConfigKeyMatchesConstraintKey) {
+        return true;
+      }
+    }
+    return false;
+  },
+  /**
+   * Determines whether the value is allowed by the constraints.
+   *
+   * @param {String} value The value.
+   * @param {Array} allowedValues The array of allowed values.
+   * @returns {Boolean} true if the value is allowed by the constraints, false otherwise.
+   * @static
+   */
+  valueIsAllowed: function(value, allowedValues) {
```

Comment:
rather than this function, which only validates string values, it might be more flexible to allow a `validate` function on the property that returns a boolean and gets checked after the type validation.

```javascript
scope: {
   type: 'string',
   validate: function(value) {
      supportedValues = ['past', 'future', 'all'];
      return supportedValues.includes(value);
   },
   optional: true
}
```